alf/utils/plot_tb_curves.py

```diff
@@ -229,31 +249,99 @@ def y_label(self):
 
 
+class MeanCurveGroupReader(object):
+    r"""Group several ``MeanCurveReader`` results. A ``MeanCurveGroupReader`` is
+    suitable for one method on multiple tasks, each task with multiple runs.
+    To aggregate across tasks, each task must be provided with a performance
+    range :math:`(y_0, y_1)` that will be used to normalize performance for that
+    task as :math:`\frac{y - y_0}{y_1 - y_0}`.
+    """
+
+    def __init__(self, mean_curve_readers, task_performance_ranges, name):
+        """
+        Args:
+            mean_curve_readers (list[MeanCurveReader]): a list of
+                ``MeanCurveReader`` of multiple tasks for one method.
+            task_performance_ranges (list[tuple(float)]): a list of tuples, where
+                each tuple is a pair of floats used for normalizing the corresponding
+                task.
+            name (str): name of the method
+        """
+
+        def _normalize(y, y0, y1):
+            return (y - y0) / (y1 - y0)
+
+        assert len(mean_curve_readers) == len(task_performance_ranges)
+        curves = [reader() for reader in mean_curve_readers]
+        y, min_y, max_y = [], [], []
+
+        for c, (y0, y1) in zip(curves, task_performance_ranges):
+            assert len(c.x) == len(curves[0].x)
+            y.append(_normalize(c.y, y0, y1))
+            min_y.append(_normalize(c.min_y, y0, y1))
+            max_y.append(_normalize(c.max_y, y0, y1))
+
+        self._mean_curve = MeanCurve(
+            x=curves[0].x,
+            name=curves[0].name,
+            y=np.mean(y, axis=0),
+            min_y=np.mean(min_y,
+                          axis=0),  # TODO: might have a better way than mean()
+            max_y=np.mean(max_y, axis=0))
+
+        self._x_label = mean_curve_readers[0].x_label
+        self._name = name
+
+    @property
+    def x_label(self):
+        return self._x_label
+
+    @property
+    def y_label(self):
+        return "Normalized Score"
+
+    @property
+    def name(self):
+        return self._name
+
+    def __call__(self):
+        return self._mean_curve
+
+
 class CurvesPlotter(object):
     """Plot several MeanCurves in a figure. The curve colors will form
     a cycle over 10 default colors.
     """
     _COLORS = ['C%d' % i for i in range(10)]
-    _LINE_WIDTH = 2
 
     def __init__(self,
                  mean_curves,
+                 y_clipping=None,
                  x_range=None,
                  y_range=None,
-                 y_clipping=None,
                  x_label=None,
                  y_label=None,
-                 x_scaled_and_aligned=True,
+                 x_scaled_and_aligned=False,
+                 figsize=(4, 4),
+                 dpi=100,
+                 linestyle='-',
+                 linewidth=2,
+                 alpha=0.3,
+                 bg_color='#EEEEEE',
+                 grid_color='#DEDEDE',
+                 legend_kwargs=dict(loc="best"),
                  title=None):
-        """
+        r"""
```

Comment:
Add simple description what the plot looks like.

Reply:
added